webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/UserManagementApi.scala

```diff
@@ -628,4 +689,29 @@ class UserManagementApiImpl(
       }
     }
   }
+
+  private def getDisplayTenants(user: RudderUserDetail):                         String = {
+    getDisplayTenants(Right(user.nodePerms))
+  }
+  private def getDisplayTenants(userSession: UserSession):                       String = {
+    // assume user logged in when tenants features did not exist
+    getDisplayTenants(
+      Left(userSession.tenants.getOrElse(NodeSecurityContext.All.value))
+    )
+  }
+  private def getDisplayTenants(nodePerms: Either[String, NodeSecurityContext]): String = {
+    nodePerms.fold(identity, _.value) match {
+      case tenantsTagsRegex(r) => r
+      case s                   => s
+    }
+  }
+}
+
+object UserManagementApiImpl {
+  // Tenants value encoded in user session of form "tags:[...]" needs to be displayed as list (see .value)
+  private val tenantsTagsRegex = """^tags:\[(.*)\]$""".r
```

Comment:
I don't think you need that, nor you want that. It creates a coupling between rudder-core and something that is out of rudder core, in the tenant plugin. The boundary dependencies need to go the other way around. 

It will be fragile and will need synchronisation between the plugin and rudder, when it was designed to not need one. 

From Rudder, you need to distinguish the case where the plugin is there, or not. If the plugin is not there, then there is only two possibilities: all or none. 
If the plugin is there, then it is *its* responsibility to translate what something is (and even give context: it's likely you don't want to display an ID but the human name of the tenant with the short description on hover or something like that). 

So, from an arch point of view, it seems that: 
- you need in Rudder something that tells you if the tenant plugin is enabled or not. I'm not sûr we have it, but if not I should be possible to model it on the old user extended right way

Then, you need to have a service that from a tenant serialisation form gives you the current (ie, with regards to the current context of plugins) value of tenant right, and from that you will do whatever you want in the api. 
That service will have a default implementation in rudder core which only says all or none, and a better implementation added in the plugin which says none, all, or a list of tenants. It's classicaly implemented with;

``` 
trait MyService { def giveMeWhatINeed(s: String) : IOResult[XXX]
object MyCoreService { def giveMeWhatINeed(s: String) : IOResult[XXX] = all or none }
class MyServiceProvider {
  val myService = Ref.make(MyCoreService)
}
// and in RudderConfig
val myServiceProvider = new MyServiceProvider(...) 
// + expose it in RudderConfig API

// and in tenant plugin
class MyBetterService(dependencies) {   def giveMeWhatINeed(s: String) : IOResult[XXX] = some logic and retrival }
// and in tenant plugin init
RudderConfig.myServiceProvider.myService.setTo(new MyBetterService(dependencies))

Reply:
:+1: Indeed, the table should display the current tenants of the user.

In fact we already have the `nodePerms: NodeSecurityContext` of the user as it is parsed from the `users.xml` file, but I just didn't realize I could use it instead of the regex... :sweat_smile: 

So, instead of using the session data for displaying the tenants, I could just use the file data, and it seems consistent with the existing model of tenants for a `RudderUserDetail`.

For the record, I initially thought it was similar to user permissions, that we need to display from previous session data for externally provided users, because we don't have any other way... In the case of tenants, it appears we can always display from the declarations in the users file, and `all` if the user is not in the file